### DISCUSSION TITLE
Exec into coredns pod

### DISCUSSION THREAD

**Discussion by ssmntn210** [coredns/coredns#5264] [Q&A] [answered]
Hi everyone, 
I wanted to exec into the cordons pod in order to use the top and tcpdump command (the first to get instantaneous CPU stats and the second one to get the DNS resolution requests to corendns). As per the issue, **https://github.com/coredns/coredns/issues/3533**, the image does not contain a shell, and I'm getting the same error:
`exec: "sh": executable file not found in $PATH: unknown`
Any way to workaround this, and get to use the top and tcpdump for coredns? 
Thanks in advance!

**Comment by mqasimsarfraz** [comment]
In case your cluster has [Ephemeral Containers](https://kubernetes.io/docs/concepts/workloads/pods/ephemeral-containers/) enabled, you should be able to use any image to inspect your running pod e.g: 
```
kubectl debug -it -n kube-system --image=busybox coredns-64897985d-4l8h8 -- sh
```
The feature is available as beta in Kubernetes 1.23

**Reply by ssmntn210** [reply]
Hi Qasim, using Kubernetes 1.14, and upgrading it will create some issues, so won't be able to use it..
Thanks for the solution!

**Comment by johnbelamaric** [comment, marked as answer]
In that case you'll need to login to the node, and use top and tcpdump on the node rather than in the container. You should be able to switch into the network namespace of the coredns process from your root shell, and then tcpdump the interface from there.

Not sure what you want with "top". In the process namespace, there is only one process (coredns). CPU and memory status should be available from your metrics server, if you have one.

**Reply by ssmntn210** [reply]
Yep, got it. Using top command from node works.
Thanks a lot!
